docs/02_cleos/03_command-reference/get/kv_table.md

```diff
@@ -12,48 +12,393 @@ Retrieves the contents of a database kv_table
+```
+```json
+{
+    "rows": [
+        "000000000000600e3d010000000000000004626f6261000000"
+    ],
+    "more": false,
+    "next_key": ""
+}
 ```
 
-Point query to return the row that matches eosio name key `boba` from kv_table named `kvtable` owned by `contr_acct` account using kv_table index `primarykey`:
+Point query to return the row that matches decimal key `1` from kv_table named `kvtable` owned by `contr_acct` account using kv_table index `foo` (note that multiple rows could have resulted since `foo` is a non-unique secondary index):
+```sh
+cleos get kv_table --encode-type dec -i 1 contr_acct kvtable foo
+```
+```json
+{
+    "rows": [
+        "000000000000600e3d010000000000000004626f6261"
+    ],
+    "more": false,
+    "next_key": ""
+}
+```
+
+Range query to return all rows that match eosio name keys between `bobd` (inclusive) and `bobh` (exclusive) from kv_table named `kvtable` owned by `contr_acct` account using kv_table index `primarykey`:
+```sh
+cleos get kv_table --encode-type name -L bobd -U bobh contr_acct kvtable primarykey -b
+```
+```json
+{
+    "rows": [
+        "000000000000900e3d040000000000000004626f6264000000",
+        "000000000000a00e3d050000000000000004626f6265000000",
+        "000000000000b00e3d060000000000000004626f6266000000",
+        "000000000000c00e3d070000000000000004626f6267000000"
+    ],
+    "more": false,
+    "next_key": ""
+}
+```
+
+Range query to return all rows (in reverse order) that match eosio name keys between `bobd` (exclusive) and `bobh` (inclusive) from kv_table named `kvtable` owned by `contr_acct` account using kv_table index `primarykey`:
+```sh
+cleos get kv_table --encode-type name -L bobd -U bobh contr_acct kvtable primarykey -b -r
+```
+```json
+{
+    "rows": [
+        "000000000000d00e3d080000000000000004626f6268000000",
+        "000000000000c00e3d070000000000000004626f6267000000",
+        "000000000000b00e3d060000000000000004626f6266000000",
+        "000000000000a00e3d050000000000000004626f6265000000"
+    ],
+    "more": false,
+    "next_key": ""
+}
+```
+
+Range query to return all rows starting from eosio name key `bobg` (inclusive) from kv_table named `kvtable` owned by `contr_acct` account using kv_table index `primarykey`:
+```sh
+cleos get kv_table --encode-type name -L bobg contr_acct kvtable primarykey -b
+```
+```json
+{
+    "rows": [
+        "000000000000c00e3d070000000000000004626f6267000000",
+        "000000000000d00e3d080000000000000004626f6268000000",
+        "000000000000e00e3d090000000000000004626f6269000000",
+        "000000000000f00e3d0a0000000000000004626f626a000000"
+    ],
+    "more": false,
+    "next_key": ""
+}
+```
+
+Range query to return all rows (in reverse order) starting from eosio name key `bobg` (exclusive) from kv_table named `kvtable` owned by `contr_acct` account using kv_table index `primarykey`:
+```sh
+cleos get kv_table --encode-type name -L bobg contr_acct kvtable primarykey -b -r
```

Comment:
missing -U bobj

Reply:
Is it really required if -r is specified? I wanted to highlight an example where -U is not specified, in which case the reverse iterator will start from the last row. Is this correct? I should correct the description to read "start from the last row until eosio name key \`bobg\` (exclusive)", though.